x-pack/legacy/plugins/apm/server/routes/settings.ts

```diff
@@ -4,220 +4,189 @@
-          _debug: Joi.bool()
-        },
-        payload: agentConfigPayloadValidation
-      },
-      tags: ['access:apm']
-    },
-    handler: async req => {
-      const setup = await setupRequest(req);
-      const configuration = req.payload as AgentConfigurationIntake;
-      return await createConfiguration({
-        configuration,
-        setup
-      }).catch(defaultErrorHandler);
-    }
-  });
-
-  // update configuration
-  server.route({
-    method: 'PUT',
-    path: `/api/apm/settings/agent-configuration/{configurationId}`,
-    options: {
-      validate: {
-        query: {
-          _debug: Joi.bool()
-        },
-        payload: agentConfigPayloadValidation
-      },
-      tags: ['access:apm']
-    },
-    handler: async req => {
-      const setup = await setupRequest(req);
-      const { configurationId } = req.params;
-      const configuration = req.payload as AgentConfigurationIntake;
-      return await updateConfiguration({
-        configurationId,
-        configuration,
-        setup
-      }).catch(defaultErrorHandler);
-    }
-  });
-
-  // Lookup single configuration
-  [
-    '/api/apm/settings/agent-configuration/search',
-    '/api/apm/settings/cm/search' // backward compatible api route for apm-server
-  ].forEach(path => {
-    server.route({
-      method: 'POST',
-      path,
-      options: {
-        validate: {
-          query: {
-            _debug: Joi.bool()
-          }
-        },
-        tags: ['access:apm']
-      },
-      handler: async (req, h) => {
-        interface Payload {
-          service: {
-            name: string;
-            environment?: string;
-          };
-        }
-
-        await createApmAgentConfigurationIndex(server);
-
-        const setup = await setupRequest(req);
-        const payload = req.payload as Payload;
-        const serviceName = payload.service.name;
-        const environment = payload.service.environment;
-        const config = await searchConfigurations({
-          serviceName,
-          environment,
-          setup
-        });
-
-        if (!config) {
-          return h.response().code(404);
-        }
-
-        return config;
-      }
+  },
+  handler: async (req, { body }, h) => {
+    const [setup] = await Promise.all([
+      setupRequest(req),
+      createApmAgentConfigurationIndex(core.http.server)
+    ]);
+
+    const config = await searchConfigurations({
+      serviceName: body.service.name,
+      environment: body.service.environment,
+      setup
     });
-  });
-}
+
+    if (!config) {
+      return h.response().code(404);
+    }
+
+    return config;
+  }
+}));
+
+export const agentConfigurationSearchRoute = createRoute(core => ({
+  ...searchRoute(core),
+  path: '/api/apm/settings/agent-configuration/search'
+}));
+
+// backward compatible api route for apm-server
+export const legacyAgentConfigurationSearchRoute = createRoute(core => ({
+  ...searchRoute(core),
+  path: '/api/apm/settings/cm/search'
+}));
```

Comment:
APM Server ended up releasing with the correct route so this can be removed without problems.

Reply:
nice! will remove.